nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AddPermission.java

```diff
@@ -81,87 +98,107 @@
     protected String email;
 
     @Param(name = "permission", description = "ACE permission.")
-    String permission;
+    protected String permission;
 
     @Param(name = "acl", required = false, values = { ACL.LOCAL_ACL }, description = "ACL name.")
-    String aclName = ACL.LOCAL_ACL;
+    protected String aclName = ACL.LOCAL_ACL;
 
     @Param(name = "begin", required = false, description = "ACE begin date.")
-    Calendar begin;
+    protected Calendar begin;
 
     @Param(name = "end", required = false, description = "ACE end date.")
-    Calendar end;
+    protected Calendar end;
 
     @Param(name = "blockInheritance", required = false, description = "Block inheritance or not.")
-    boolean blockInheritance = false;
+    protected boolean blockInheritance = false;
 
     @Param(name = "notify", required = false, description = "Notify the user or not")
-    boolean notify = false;
+    protected boolean notify = false;
 
     @Param(name = "comment", required = false, description = "Comment")
-    String comment;
+    protected String comment;
 
     @OperationMethod(collector = DocumentModelCollector.class)
     public DocumentModel run(DocumentModel doc) {
+        validateParameters();
```

Comment:
Why not calling `validateParameters` only one time in `addPermission`?

Reply:
It could be done, in fact.
The rational used was, in first place, to separate the logic and then for semantic reasons. 
I.e., addPermission will just add the permission to the document and not validate the operation parameters.

Reply:
I do not have any strong feelings against it.